omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -514,6 +515,20 @@ def generate_config(node_name)
         PrivateChef["default_listen_address"] = "::"
       end
 
+      # Transition Solr memory and JVM settings from OSC11 to Chef 12.
+      if PrivateChef['opscode_solr'].has_key? 'heap_size'
+        PrivateChef['opscode_solr4']['heap_size'] ||= PrivateChef['opscode_solr']['heap_size']
+        PrivateChef['opscode_solr'].delete 'heap_size'
+      end
+      if PrivateChef['opscode_solr'].has_key? 'new_size'
+        PrivateChef['opscode_solr4']['new_size'] ||= PrivateChef['opscode_solr']['new_size']
+        PrivateChef['opscode_solr'].delete 'new_size'
+      end
+      if PrivateChef['opscode_solr'].has_key? 'java_opts'
+        PrivateChef['opscode_solr4']['java_opts'] ||= PrivateChef['opscode_solr']['java_opts']
+        PrivateChef['opscode_solr'].delete 'java_opts'
+      end
+
```

Comment:
Looks good, but could be simplified slightly: 
```
old_config = PrivateChef['opscode_solr'] || {}
[ 'heap_size', 'new_size', 'java_opts' ].each do |key| 
  if old_config.has_key? key
    PrivateChef['opscode_solr4'][key] = old_config[key]
   old_config.delete key 
  end 
end
```

Note that while this will remove the config entries and prevent them from being saved to the node or chef-server-running.json, they will still live on in chef-server.rb / private-chef.rb